### PR TITLE
Enable xUnit verbose logging on System.IO.Tests

### PR DESCRIPTION
**Pull request by jozkee**
Contributes to https://github.com/dotnet/runtime/issues/100558 in an attempt to visualize which test is causing OOM (error 137).

### CODE REVIEW

**Comment by carlossanlop** [comment]
What will be the main difference?

There's also the Net5Compat tests failing with the same error. Are those handled with a different csproj? I don't remember. If they are, then they will need the same property added.

**Comment by jozkee** [comment]
> What will be the main difference?

This is akin to use `dotnet test --logger "console;verbosity=detailed"`.

> Net5Compat 

 System.IO.Net5Compat.Tests does not exist in main.

**Comment by jozkee** [comment]
@carlossanlop here's the resulting console log: 
https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-107149-merge-28a97be8c4ad43219c/System.IO.Tests/1/console.274fae68.log?helixlogtype=result

**Review by carlossanlop — APPROVED** [review]

**Comment by carlossanlop** [comment]
> System.IO.Net5Compat.Tests does not exist in main, it was removed in #57735

Ok good to know. So when we find the offending test, we can hopefully do something similar servicing branches to unblock the Net5Compat failing tests.

**Comment by jozkee** [comment]
Trying https://github.com/dotnet/runtime/pull/107163 before merging this.

**Comment by jozkee** [comment]
I'm still seeing OOM errors after https://github.com/dotnet/runtime/pull/107163.

https://dev.azure.com/dnceng-public/public/_build/results?buildId=793852&view=ms.vss-test-web.build-test-results-tab&runId=20432094&resultId=211296

https://dev.azure.com/dnceng-public/public/_build/results?buildId=793828&view=ms.vss-test-web.build-test-results-tab&runId=20430560&resultId=220504

**Comment by jozkee** [comment]
/ba-g error on `WasmTestOnChrome-ST-System.Runtime.Tests.WorkItemExecution` is https://github.com/dotnet/runtime/issues/103347.

**Comment by jkotas** [comment]
> I'm still seeing OOM errors after https://github.com/dotnet/runtime/pull/107163.

#107163 was merged into `main`. The two failures you have shared are from `release/9.0`. Do you still see failures in `main`?

**Comment by jozkee** [comment]
> Do you still see failures in main?

No, latest failure in main was https://dev.azure.com/dnceng-public/public/_build/results?buildId=793300&view=ms.vss-test-web.build-test-results-tab&runId=20409750&resultId=211437. 

That happened before merging https://github.com/dotnet/runtime/pull/107163. Thanks for catching that, @jkotas.

**Pull request merged**
